### DISCUSSION TITLE
How does host detect if plugin accepts MIDI or is straight Audio?

### DISCUSSION THREAD

**Discussion by sagantech** [free-audio/clap#167] [Q&A] [answered]
Any help greatly appreciated.

**Comment by robbert-vdh** [comment, marked as answer]
You check the [audio-ports](https://github.com/free-audio/clap/blob/main/include/clap/ext/audio-ports.h) and [note-ports](https://github.com/free-audio/clap/blob/main/include/clap/ext/note-ports.h) extensions.

**Reply by sagantech** [reply]
Thank you. Perfect.

**Reply by baconpaul** [reply]
the clap-info tool (https://github.com/free-audio/clap-info) is a command line example of a scanner which dumps all the config information about a particular clap, if that helps.
